gradle/java-library.gradle

```diff
@@ -3,6 +3,7 @@ apply plugin: "java-library"
 apply from: "$rootDir/gradle/test-launcher.gradle"
 
 if (!base.archivesName.get().startsWith("mockito-")) {
+    logger.warn("Project module should be prefixed with `mockito-` : ${project.projectDir})")
```

Comment:
Let's remember to throw an exception in the future here to keep things consistent (I assume we can't throw it yet until #3453 hasn't merged yet?)

Reply:
Yes that's my take.
I agree that we could fail the build if that's the case.